src/qibolab/serialize.py

```diff
@@ -39,19 +40,45 @@ def load_qubits(runcard: dict) -> Tuple[QubitMap, QubitPairMap]:
         pair = tuple(sorted(pair))
         pairs[pair] = QubitPair(qubits[pair[0]], qubits[pair[1]])
 
-    # register single qubit native gates to ``Qubit`` objects
+    return qubits, pairs
+
+
+def load_couplers(runcard: dict) -> CouplerMap:
+    """Load couplers and pairs from the runcard.
+
+    Uses the native gate and characterization sections of the runcard
+    to parse the :class:`qibolab.qubits.Qubit` and :class:`qibolab.qubits.QubitPair`
+    and the the :class:`qibolab.coupler.Coupler`
+    objects.
+    """
+    couplers = {c: Coupler(c, **char) for c, char in runcard["characterization"]["coupler"].items()}
+
+    return couplers
+
+
+# This creates the compiler error
```

Comment:
Does this still create an error? I did not test.

Reply:
As far as I remember this was related to something not getting loaded properly just using `load_qubits`.

Reply:
I think it was related to the step where I assign couplers to qubits that it's done in in between the` load_qubit` and `register_gates`